### PR TITLE
Update README to include installation instructions and an API summary

### PR DESCRIPTION
**Pull request by as2388**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,17 +1,83 @@
...
+data =  # TODO: assign iterable of TracedData objects
+
+with open("output.json", "w") as f:
+    TracedDataJsonIO.export_traced_data_iterable_to_json(data, f)
+    
+with open("input.json", "r") as f:
+    data = TracedDataJsonIO.import_json_to_traced_data_iterable(f)
+```
+
+#### Cleaning
```
Comment (lukechurch) [inline README.md]: This is a layering violation - this structure knows nothing about cleaning - could we move this to the demonstrator repo readme?

File: README.md [hunk 2]
```diff
@@ -1,17 +1,83 @@
...
+All reusable cleaning functions may be used by importing functionality from `core_data_modules.cleaners`.
+For example:
+```python
+from core_data_modules.cleaners.english.demographic_cleaner import DemographicCleaner
+
+DemographicCleaner.clean_gender("woman")  # 'F'
+DemographicCleaner.clean_gender("aoeu")  # None
+```
+Note that cleaners which fail to assign a cleaned value return `None`.
```
Comment (lukechurch) [inline README.md]: How does None serialise to JSON?

Comment (as2388) [reply README.md]: null

(which deserializes back to None)

File: README.md (hunk 1 above)
Comment (as2388) [reply README.md]: I don't understand - the cleaners in this repository. Not showing how to use them here would be like not showing how to use TracedData.

Comment (lukechurch) [reply README.md]: Ahh, I see.

In which case, I think we should make just make the IO a subsection of TracedData (all of the IO functions use trace data IIRC?).

Comment (as2388) [reply README.md]: That makes more sense. Updated as requested.

**Pull request merged**